src/_backend_gdk.c

```diff
@@ -52,7 +52,15 @@ static PyObject *pixbuf_get_pixels_array(PyObject *self, PyObject *args)
 
     /* the array holds a ref to the pixbuf pixels through this wrapper*/
     Py_INCREF(py_pixbuf);
-    PyArray_SetBaseObject(array, (PyObject *)py_pixbuf);
+#if NPY_API_VERSION >= 0x00000007
+    if (PyArray_SetBaseObject(array, (PyObject *)py_pixbuf) == -1) {
+        Py_DECREF(py_pixbuf);
+        Py_DECREF(array);
+        return NULL;
+    }
+#else
+    PyArray_BASE(array) = py_pixbuf;
```

Comment:
> On 16 Sep 2015, at 17:18, mdehoon <notifications@github.com> wrote:
> 
> In src/_backend_gdk.c <https://github.com/matplotlib/matplotlib/pull/5080#discussion_r39634884>:
> 
> > @@ -52,7 +52,15 @@ static PyObject *pixbuf_get_pixels_array(PyObject *self, PyObject *args)
> >  
> >      /* the array holds a ref to the pixbuf pixels through this wrapper*/
> >      Py_INCREF(py_pixbuf);
> > -    PyArray_SetBaseObject(array, (PyObject *)py_pixbuf);
> > +#if NPY_API_VERSION >= 0x00000007
> > +    if (PyArray_SetBaseObject(array, (PyObject *)py_pixbuf) == -1) {
> > +        Py_DECREF(py_pixbuf);
> > +        Py_DECREF(array);
> > +        return NULL;
> > +    }
> > +#else
> > +    PyArray_BASE(array) = py_pixbuf;
> Yes, because PyArray_SetBaseObject expects a PyObject pointer as its second argument.
> 
I guess the question is whether we need to do the cast in the #else branch too. It’s not required in C, but assigning to a pointer to a different type can cause a compiler warning, so it would be slightly cleaner to cast in that branch too.